### PR TITLE
Coverage

### PR DESCRIPTION
**Pull request by wallner**
Enable the generation of code coverage information and transmit the information to both codacy.com and circleci.

### CODE REVIEW

**Comment by wallner** [comment]
Please merge pr #113 before this one. I will rebase it if necessary.

**Comment by tkrille** [comment]
Please rebase now.

**Comment by wallner** [comment]
done.

File: pom.xml [hunk 1]
```diff
@@ -325,7 +327,7 @@
                 <artifactId>maven-surefire-plugin</artifactId>
                 <version>2.19</version>
                 <configuration>
-                    <argLine>-XX:-UseSplitVerifier</argLine>
+                    <argLine>${argLine}</argLine>
```
Comment (tkrille) [inline pom.xml:330]: Why this variable? What is its content and where does it come from?

File: circle.yml [hunk 2]
```diff
@@ -5,9 +5,18 @@ machine:
...
     - mvn -U dependency:resolve dependency:resolve-plugins
+
 test:
   override:
     - mvn verify
+  post:
+    - mkdir -p $CIRCLE_TEST_REPORTS/junit/
+    - find . -type f -regex ".*/target/.*-reports/.*xml" -exec cp {} $CIRCLE_TEST_REPORTS/junit/ \;
+    - cp -r target/coverage-reports/jacoco/ $CIRCLE_ARTIFACTS
+    - curl http://www.jpm4j.org/install/script > jpmInstall.sh
```
Comment (tkrille) [inline circle.yml:16]: I'm just wondering whether we could install `jpm` and `codacy-coverage-reporter` in the dependency phase and change the caching settings in a way, that both will be cached.

File: pom.xml (hunk 1 above)
Comment (wallner) [inline pom.xml:330]: It comes from jacoco and orchestrates the javaagent.

File: circle.yml (hunk 2 above)
Comment (wallner) [inline circle.yml:16]: This is the way documented by codacy for circleci. I would like to use their default first. I'm happy to review your PR once this one is merged, though.

**Comment by tkrille** [comment]
Alright, thanks for the clarifications.

**Pull request merged**